src/plugins/calculate/calculate.c

```diff
@@ -0,0 +1,331 @@
+		PNElem e2 = nextVal(stackPtr);
+		if(e1.op == VAL && e2.op == VAL)
+		{
+			switch(stackPtr->op)
+			{
+
+				case ADD:
+					stackPtr->value = e1.value + e2.value; 
+					stackPtr->op = VAL;
+					break;
+				case SUB:
+					stackPtr->value = e1.value - e2.value; 
+					stackPtr->op = VAL;
+					break;
+				case DIV:
+					if(e2.value == 0)
+					{
+						result.op = ERROR;
+						return result;
+					}
+					stackPtr->value = e1.value / e2.value; 
+					stackPtr->op = VAL;
+					break;
+				case MUL:
+					stackPtr->value = e1.value * e2.value; 
+					stackPtr->op = VAL;
+					break;
+				default:
+					break;
+			}
+		}
+		else
+		{
+			return result;
+		}
+	}
+	if(stackPtr->op != VAL)
+		result.op = ERROR;
+	else
+		result.op = RES;
+	result.value = stackPtr->value;
+	return result;
+}
+static PNElem parsePrefixString(const char *prefixString, KeySet *ks, Key *parentKey)
+{
+	char *regexString = "((([[:alnum:]]*/)+[[:alnum:]]+))|([-+/<>=!{*])";
+	char *ptr = (char *)prefixString;
+	regex_t regex;
+	Key *key;
+	//PNElem stack[100];
+	
+	PNElem *stack = malloc(3*sizeof(PNElem));
+
+	PNElem *stackPtr = stack;
+	PNElem result;
+	Operation resultOp = ERROR;
+	result.op = ERROR;
+	int ret;
+	if((ret = regcomp(&regex, regexString, REG_EXTENDED|REG_NEWLINE)))
+	{
+		ksDel(ks);
+		return result;
+	}
+	regmatch_t match;
+	int nomatch;
+	int start;
+	int len;
+	char *searchKey = NULL;
+	while(1)
+	{
+		nomatch = regexec(&regex, ptr, 1, &match, 0);
+		if(nomatch)
+		{
+			break;
+		}
+		len = match.rm_eo - match.rm_so;
+		start = match.rm_so + (ptr - prefixString);
+		if(len == 1)
+		{
+			switch(prefixString[start])
+			{
+				case '+':
+					stackPtr->op = ADD;
+					++stackPtr;
+					break;
+				case '-':
+					stackPtr->op = SUB;
+					++stackPtr;
+					break;
+				case '/':
+					stackPtr->op = DIV;
+					++stackPtr;
+					break;
+				case '*':
+					stackPtr->op = MUL;
+					++stackPtr;
+					break;
+				case '=':
+					if(resultOp == LT)
+						resultOp = LE;
+					else if(resultOp == GT)
+						resultOp = GE;
+					else if(resultOp == ERROR)
+						resultOp = EQU;
+					else if(resultOp == EQU)
+						resultOp = EQU;
+					else if(resultOp == NOT)
+						resultOp = NOT;
+					break;
+				case '<':
+					resultOp = LT;
+					break;
+				case '>':
+					resultOp = GT;
+					break;
+				case '!':
+					resultOp = NOT;
+					break;
+				default:
+					ELEKTRA_SET_ERRORF(122, parentKey, "%c isn't a valid operation", prefixString[start]);
+					regfree(&regex);
+					if(searchKey)
+						free(searchKey);
+					free(stack);
+					ksDel(ks);
+					return result;	
+					break;
+			}
+		}
+		else
+		{
+			ksRewind(ks);
+			searchKey = realloc(searchKey, len+2+strlen(keyName(parentKey)));
+			strcpy(searchKey, keyName(parentKey));
+			strcat(searchKey, "/");
+			strncat(searchKey, ptr+1, len);
+			key = ksLookupByName(ks, searchKey, 0);
+			if(!key)
+			{
+				ELEKTRA_SET_ERRORF(124, parentKey, "Couldn't find Key %s", searchKey);
```

Comment:
Give more context to the user: Why was the key searched (which meta key, which operation,...)